lib/config.js

```diff
@@ -229,13 +228,44 @@ function validateEnvConfig (prefix) {
   }
 }
 
+function deepFreeze (o) {
+  Object.freeze(o)
+
+  Object.getOwnPropertyNames(o).forEach(function (prop) {
+    if (o.hasOwnProperty(prop) &&
```

Comment:
Since you're iterating `getOwnPropertyNames`, isn't testing `hasOwnProperty` redundant?